### PR TITLE
Budget main screen

### PR DESCRIPTION
**Pull request by dwulfse**

### CODE REVIEW

File: components/Budget/BudgetInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return;
+        }
+    }, []);
+
+    return (
+        <TouchableOpacity
+            onPress={loading ? () => {} : props.onClick}
+            className="flex px-5 py-2 border-2 rounded-xl my-2"
+        >
+            {loading ? (
```
Comment (DanWilkins2107) [inline components/Budget/BudgetInfo.jsx]: Atm, getting a budget that is permanently loading when it doesn't properly exist, wonder if we need some logic to remove it from the array and just have an alert of "could not fetch all budgets?"

File: components/Budget/BudgetInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                "Error occurred when fetching budgets.",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+            return;
+        }
+    }, []);
+
+    return (
+        <TouchableOpacity
```
Comment (DanWilkins2107) [inline components/Budget/BudgetInfo.jsx:11]: Wonder whether this should have a background/format more similar to the production buttons

File: components/Budget/BudgetInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            onPress={loading ? () => {} : props.onClick}
+            className="flex px-5 py-2 border-2 rounded-xl my-2"
+        >
+            {loading ? (
+                <LoadingWheel size="large" color="#0000ff" />
+            ) : (
+                <>
+                    <View className="flex flex-row justify-between">
+                        <Text className="text-2xl font-bold">{budget.name}</Text>
+                        <FontAwesomeIcon
```
Comment (DanWilkins2107) [inline components/Budget/BudgetInfo.jsx]: Icon looks out of line with the name of the budget, tailwind equivalent of align-items: center should sort (items-center?)

File: components/Budget/BudgetInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                            icon={icon({ name: "chevron-right" })}
+                            size={20}
+                            className="py-2z"
+                        />
+                    </View>
+                    <Text className="text-lg font-light">Budget: £{budget.budget}</Text>
+                    <Text className="text-lg font-light">
+                        Participants: {participants ? participants : "no participants"}
```
Comment (DanWilkins2107) [inline components/Budget/BudgetInfo.jsx]: Relating to the comment earlier about similarity to production button, I wonder whether the profile pics would be better

File: components/Budget/ViewBudgetModal.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import { ref, onValue } from "firebase/database";
+import { AlertContext } from "../Alert/AlertProvider";
+import { ModalContext } from "../Modal/ModalProvider";
+import { firebase_auth, firebase_db } from "../../firebase.config";
+import BudgetInfo from "./BudgetInfo.jsx";
+import { icon } from "@fortawesome/fontawesome-svg-core/import.macro";
+
+export default function ViewBudgetModal({ productionCode }) {
+    const { setAlert } = useContext(AlertContext);
+    //const { setModal } = useContext(ModalContext);
```
Comment (DanWilkins2107) [inline components/Budget/ViewBudgetModal.jsx]: Delete unnecessary comments

File: components/Budget/CreateBudgetModal.jsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { useState, useEffect, useContext } from "react";
 import { View, Text } from "react-native";
-import { set, ref, onValue } from "firebase/database";
+import { set, ref, push, onValue } from "firebase/database";
```
Comment (DanWilkins2107) [inline components/Budget/CreateBudgetModal.jsx]: push not needed

File: components/Budget/ViewBudgetModal.jsx (hunk 5 above)
Comment (DanWilkins2107) [reply components/Budget/ViewBudgetModal.jsx]: + delete imports

File: components/Budget/ViewBudgetModal.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    }, []);
+
+    return (
+        <View className="flex p-3">
+            <Text className="text-3xl font-extrabold text-center mb-3">Select Budget</Text>
+            <ScrollView className="flex flex-col h-96">
+                {budgets.length > 0 ? (
+                    budgets.map((budget, i) => {
+                        return (
+                            <BudgetInfo
```
Comment (DanWilkins2107) [inline components/Budget/ViewBudgetModal.jsx:57]: Think I put the comment in the wrong place earlier with the having a permanent loading, basically getting that there is a loading screen permanently and it's not obvious to the user as to why

**Review by DanWilkins2107 — CHANGES_REQUESTED** [review]

File: components/Budget/CreateBudgetModal.jsx (hunk 6 above)
Comment (DanWilkins2107) [reply components/Budget/CreateBudgetModal.jsx]: is still there, probs added again during experimenting

File: components/Budget/ViewBudgetModal.jsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+import BudgetInfo from "./BudgetInfo.jsx";
+import LoadingWheel from "../Loading/LoadingWheel.jsx";
+
+export default function ViewBudgetModal({ productionCode }) {
+    const { setAlert } = useContext(AlertContext);
+    const [budgets, setBudgets] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const db = firebase_db;
+
+    useEffect(() => {
```
Comment (DanWilkins2107) [inline components/Budget/ViewBudgetModal.jsx:16]: Main Error: app is crashing when clicking view budget on production EI0PB9, maybe try joining the production and seeing this?

File: components/Budget/ViewBudgetModal.jsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        onValue(ref(db, `productions/${productionCode}/budgets`), async (productionSnapshot) => {
+            if (!productionSnapshot.exists()) {
+                setLoading(false);
+                return;
+            }
+            const newBudgets = await Promise.all(
+                Object.keys(productionSnapshot.val()).map(async (budget) => {
+                    console.log(budget);
```
Comment (DanWilkins2107) [inline components/Budget/ViewBudgetModal.jsx]: This should be removed

File: components/Budget/ViewBudgetModal.jsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+                            if (!budgetSnapshot.exists()) return;
+                            return budgetSnapshot.val();
+                        })
+                        .catch((error) => {
+                            console.log("error: ", error.message);
+                        });
+                })
+            );
+
+            console.log(newBudgets);
```
Comment (DanWilkins2107) [inline components/Budget/ViewBudgetModal.jsx]: Same with this

**Review by DanWilkins2107 — CHANGES_REQUESTED** [review]

File: components/Budget/ViewBudgetModal.jsx (hunk 8 above)
Comment (dwulfse) [reply components/Budget/ViewBudgetModal.jsx:16]: think its when there's a budget with no participants

Comment (dwulfse) [reply components/Budget/ViewBudgetModal.jsx:16]: sorted that

Comment (dwulfse) [reply components/Budget/ViewBudgetModal.jsx:16]: I should probably be filtering by whether the user is a member of the budget tho tbf

Comment (dwulfse) [reply components/Budget/ViewBudgetModal.jsx:16]: done that now too

**Review by DanWilkins2107 — APPROVED** [review]

**Pull request merged**